hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/namenode/ha/TestEditLogTailer.java

```diff
@@ -429,19 +432,43 @@ public void testStandbyTriggersLogRollsWhenTailInProgressEdits()
       waitForStandbyToCatchUpWithInProgressEdits(standby, activeTxId,
           standbyCatchupWaitTime);
 
+      long curTime = standby.getNamesystem().getEditLogTailer().getTimer()
+          .monotonicNow();
+      // Set TestTimer that returns (curTime + 1 sec) ms while calling
+      // timer.monotonicNow(). This returned value is used by
+      // EditLogTailer.tooLongSinceLastLoad() to determine whether
+      // EditLogTailer should roll the log. Since this test is time
+      // and resource sensitive, we are using custom Timer instance
+      // so that tooLongSinceLastLoad() always returns false
+      // until we change Timer instance in EditLogTailer.
+      standby.getNamesystem().getEditLogTailer().setTimerForTest(
+          new TestTimer(curTime + TimeUnit.SECONDS.toMillis(1)));
```

Comment:
Instead of changing the timer instance multiple times, can you store a reference to the timer, then adjust the time returned by the timer?
```
TestTimer timer = new TestTimer(...);
standby.getNamesystem().getEditLogTailer().setTimerForTest(timer);
timer.setTime(...);
...
timer.setTime(...);
```